File: Récupération 20210215/AutoUpdateService 2.0.0.23/Module/mFunction.vb
Option Strict On
Option Explicit On

Module mFunction

    Public Function StrCompTextVersions(ByVal version1 As String, ByVal version2 As String) As Integer

        StrCompTextVersions = 0

        ' 1  version1 > version2
        '-1  version1 < version2
        ' 0  version1 = version2

        If version2 = Nothing Then Return 1
        If version1 = Nothing Then Return -1

        Try

            Dim m_a() As String = version1.Split(CChar("."))
            Dim m_b() As String = version2.Split(CChar("."))

            For Index As Integer = 0 To m_a.Length - 1
                If Val(m_a(Index)) > Val(m_b(Index)) Then
                    StrCompTextVersions = 1
                    Exit For
                ElseIf Val(m_a(Index)) < Val(m_b(Index)) Then
                    StrCompTextVersions = -1
                    Exit For
                End If
            Next

        Catch ex As Exception
            Return 1
        End Try

    End Function

    Public Function IsDirectoryWritable(ByVal path As String) As Boolean
        Dim info As New IO.DirectoryInfo(path)
        Return (info.Attributes And IO.FileAttributes.ReadOnly) <> IO.FileAttributes.ReadOnly
    End Function

    Public Function ArrayStringOf(ByVal f As ArrayList, ByVal findstring As String, Optional ByVal startindex As Integer = 0) As String

        For Each g As String In f
            Dim nvIndex As Integer = f.LastIndexOf(g)
            If g.Contains(findstring) And nvIndex > startindex Then
                Return g.Replace(findstring, Nothing)
            End If
        Next

        Return Nothing

    End Function

End Module

File: JFCUpdateService/JFCUpdateService/mFunction.cs
using System;
using System.Collections;
using System.IO;
using Microsoft.VisualBasic;
using Microsoft.VisualBasic.CompilerServices;


namespace JFCUpdateService
{
    [StandardModule]
    internal sealed class mFunction
    {
        public static int StrCompTextVersions(string version1, string version2)
        {
            int result = 0;
            if (Operators.CompareString(version2, null, TextCompare: false) == 0)
            {
                return 1;
            }
            if (Operators.CompareString(version1, null, TextCompare: false) == 0)
            {
                return -1;
            }
            checked
            {
                try
                {
                    string[] array = version1.Split('.');
                    string[] array2 = version2.Split('.');
                    int num = array.Length - 1;
                    for (int i = 0; i <= num; i++)
                    {
                        if (Conversion.Val(array[i]) > Conversion.Val(array2[i]))
                        {
                            return 1;
                        }
                        if (Conversion.Val(array[i]) < Conversion.Val(array2[i]))
                        {
                            return -1;
                        }
                    }
                    return result;
                }
                catch (Exception ex)
                {
                    ProjectData.SetProjectError(ex);
                    Exception ex2 = ex;
                    result = 1;
                    ProjectData.ClearProjectError();
                    return result;
                }
            }
        }

        public static bool IsDirectoryWritable(string path)
        {
            return (new DirectoryInfo(path).Attributes & FileAttributes.ReadOnly) != FileAttributes.ReadOnly;
        }

        public static string ArrayStringOf(ArrayList f, string findstring, int startindex = 0)
        {
            IEnumerator enumerator = default(IEnumerator);
            try
            {
                enumerator = f.GetEnumerator();
                while (enumerator.MoveNext())
                {
                    string text = Conversions.ToString(enumerator.Current);
                    int num = f.LastIndexOf(text);
                    if (text.Contains(findstring) && num > startindex)
                    {
                        return text.Replace(findstring, null);
                    }
                }
            }
            finally
            {
                if (enumerator is IDisposable)
                {
                    (enumerator as IDisposable).Dispose();
                }
            }
            return null;
        }

        public static string ConvertOctets(double Octets)
        {
            if (Octets >= 1099511627776.0)
            {
                return Virgul(Octets / 1099511627776.0) + "To";
            }
            if (Octets >= 1073741824.0)
            {
                return Virgul(Octets / 1073741824.0) + "Go";
            }
            if (Octets >= 1048576.0)
            {
                return Virgul(Octets / 1048576.0) + "Mo";
            }
            if (Octets >= 1024.0)
            {
                return Virgul(Octets / 1024.0) + "Ko";
            }
            return Strings.Trim(Conversions.ToString(Octets)) + "o";
        }

        private static string Virgul(double Val_Renamed)
        {
            int num = Strings.Len(Strings.Trim(Conversions.ToString(Conversion.Int(Val_Renamed))));
            int digits = ((num < 3) ? checked(3 - num) : 0);
            return Strings.Trim(Conversions.ToString(Math.Round(Val_Renamed, digits)));
        }
    }
}